plugins/woocommerce/includes/wc-account-functions.php

```diff
@@ -29,15 +29,16 @@ function wc_lostpassword_url( $default_url = '' ) {
 
 	$wc_account_page_url    = wc_get_page_permalink( 'myaccount' );
 	$wc_account_page_exists = wc_get_page_id( 'myaccount' ) > 0;
+	$is_account_page        = get_the_permalink() === $wc_account_page_url;
```

Comment:
Thanks @rjchow, it's a good point. Now that you mention it I don't have total confidence in the solution. It is probably better to re-think a way to detect the admin screen.

The existing code (below) also assumes the My Account page has been set up in the standard way, but I can't say with confidence my change would have no impact on any customizations being made.

```
if ( $wc_account_page_exists  && ! empty( $lost_password_endpoint ) ) {
	return wc_get_endpoint_url( $lost_password_endpoint, '', $wc_account_page_url );
}
```

Reply:
@rjchow I addressed this in [4c8f9cd](https://github.com/woocommerce/woocommerce/pull/50200/commits/4c8f9cd4a5de83cbdd3e085f5ba78c35dd3f9493). 

Since I couldn't use `is_login()` (see https://github.com/woocommerce/woocommerce/pull/50200#issuecomment-2262678230), I've used an alternative mentioned in its [announcement post](https://make.wordpress.org/core/2022/09/11/new-is_login-function-for-determining-if-a-page-is-the-login-screen/). I elected to use the action detection because I understand the login page name in `$pagenow` can be commonly changed ([example](https://jetpack.com/blog/wordpress-login-url/)). I think ultimately we need to use `is_login()` but will need to get a bug fixed upstream first.